src/docs/sphinx/dev_guide/tensor_dual.rst

```diff
@@ -0,0 +1,405 @@
+.. _header-n0:
+
+============
+Tensor Class
```

Comment:
There are a lot of hand coded examples on this page.  I would prefer them to be inside a source file, that is compiled, and literalinclude'd into this page.  It is very easy for examples to get out of sync with what actually works in the code.

Reply:
I understand your point, but many of the examples in this document are of things that (intentionally) do not compile. These examples are intended to be a conceptual overview of what sort of things you can do with these classes. Perhaps we can have some additional example sources (where everything compiles and runs as expected) in the repo with links to those files in this document.